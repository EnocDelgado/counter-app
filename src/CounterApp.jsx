import PropTypes from 'prop-types'
import { useState } from 'react'

/**
 * 
 * This is a Counter App
 * @param {value} param0 
 * @returns  value that may change depending on useState
 */
export const CounterApp = ({ value }) => {

    const [counter, setCounter] = useState( value );

    const handleAdd = () => {
        setCounter( counter + 1)
    };

    const handleReset = () => {
        setCounter( value )
    };

    const handleSub = () => {
        setCounter( counter - 1)
    };

    return (
        <>
            <div className='container'>

                <div className='counter'>
                    <h1>Counter App</h1>
                    <h2>{ counter }</h2>

                    <button onClick={ handleSub }>
                        -1
                    </button>

                    <button onClick={ handleReset }>
                        Reset
                    </button>

                    <button aria-label='btn-reset' onClick={ handleAdd }>
                        +1
                    </button>
                </div>
                
            </div>
        </>
    );
};

// PropTypes allows us to define a strict value type
CounterApp.propTypes ={
    value: PropTypes.number.isRequired,
};
