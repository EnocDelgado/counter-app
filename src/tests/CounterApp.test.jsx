import { fireEvent, render, screen } from "@testing-library/react"
import { CounterApp } from "../CounterApp";


describe('Test on CounterApp', () => {

    const initialValue = 10;

    test('should do match with snapshot', () => { 

        const { container } = render(<CounterApp value={ initialValue } />);
        expect( container ).toMatchSnapshot();
    });

    test('should display initial value 100 on <CounterApp value={100} />', () => {

        render( <CounterApp value={ 100 } /> );
        expect( screen.getByText( 100 ) ).toBeTruthy();
        // expect( screen.getByRole('heading',{ level: 2}).innerHTML ).toContain('100')
    });

    test('should increment with +1 button', () => {

        render( <CounterApp value={ initialValue } /> );
        fireEvent.click( screen.getByText( '+1' ) )
        expect( screen.getByText('11') ).toBeTruthy();
    })

    test('should decrement with -1 button', () => {

        render( <CounterApp value={ initialValue } /> );
        fireEvent.click( screen.getByText( '-1' ) );
        expect( screen.getByText( '9' ) ).toBeTruthy();
    });

    test('should work reset button', () => {

        render( <CounterApp value={ 0 } /> );
        fireEvent.click( screen.getByText( '+1' ) );
        fireEvent.click( screen.getByText( '+1' ) );
        fireEvent.click( screen.getByText( '+1' ) );
        // fireEvent.click( screen.getByText('Reset') );
        fireEvent.click( screen.getByRole( 'button', { name: 'btn-reset' } ) );

        expect( screen.getByText( 0 ) ).toBeTruthy();
    });
})