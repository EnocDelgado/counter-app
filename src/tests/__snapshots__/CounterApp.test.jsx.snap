// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on CounterApp should do match with snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="counter"
    >
      <h1>
        Counter App
      </h1>
      <h2>
        10
      </h2>
      <button>
        -1
      </button>
      <button>
        Reset
      </button>
      <button
        aria-label="btn-reset"
      >
        +1
      </button>
    </div>
  </div>
</div>
`;
